site_scons/site_tools/extra/extra.py

```diff
@@ -87,7 +88,9 @@ def _preprocess_emitter(source, target, env):
                 continue
             if mod != "":
                 prefix = prefix + "_" + mod
-        target.append(prefix + base + "_pp.c")
+            newtarget = dirname + '/' + prefix + base + "_pp" + ext
+            print("newtarget = %s\n" % newtarget)
+            target.append(dirname + '/' + prefix + base + "_pp" + ext)
```

Comment:
If I've read this code right then the below would be clearer.  Does the "ext" include the "." or not as there may be a bug-fix here as well.
```suggestion
            newtarget = os.path.join(dirname, '{}{}_pp.{}'.format(prefix, base, ext))
            print("newtarget = %s\n" % newtarget)
            target.append(newtarget)
```